packages/editorconfig/src/browser/editorconfig-document-manager.ts

```diff
@@ -66,23 +72,25 @@ export class EditorconfigDocumentManager {
                     const uri = monacoEditor.uri.toString();
                     const properties = this.properties[uri];
 
-                    const edits = [];
-                    edits.push(...this.getEditsTrimmingTrailingWhitespaces(monacoEditor, properties, event.reason));
-
+                    const edits: monaco.editor.IIdentifiedSingleEditOperation[] = [];
                     const edit = this.getEditInsertingFinalNewLine(monacoEditor, properties);
-                    if (edit) {
-                        edits.push(edit);
-
-                        // get current cursor position
-                        const cursor = monacoEditor.cursor;
-
-                        // and then restore it after resolving the promise
-                        setTimeout(() => {
-                            monacoEditor.cursor = cursor;
-                        }, 0);
-                    }
-
-                    resolve(edits);
+                    return Promise.all([
+                        this.getEditsTrimmingTrailingWhitespaces(monacoEditor, properties, event.reason),
+                        this.formatDocument(monacoEditor)
+                    ]).then(() => {
+                        if (edit) {
+                            edits.push(edit);
```

Comment:
`this.getEditsTrimmingTrailingWhitespaces()` also returns "edit" that should be pushed in to `edits`. I believe it is missing in the change.

Reply:
Done, adjusted the return value